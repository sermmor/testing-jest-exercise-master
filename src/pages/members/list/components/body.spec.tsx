import * as React from 'react';
import { Body } from './body';
import { Member } from '../viewModel';
import { shallow } from 'enzyme';

describe('pages/members/list/components/body test', () => {
    it('should be a valid body rendered', () => {
        const membersEntry : Member[] = [
            {
                id: 1,
                name: 'test field login',
                avatarUrl: 'http://testavatar.png'
            },
            {
                id: 2,
                name: 'test field login (1)',
                avatarUrl: 'http://testavatar(1).png',
            },
            {
                id: 3,
                name: 'test field login (2)',
                avatarUrl: 'http://testavatar(2).png',
            },
        ];

        const component = shallow(<Body members={membersEntry}/>);

        expect(component).toMatchSnapshot();
    });
});
