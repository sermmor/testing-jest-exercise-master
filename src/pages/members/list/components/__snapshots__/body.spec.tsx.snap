// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/members/list/components/body test should be a valid body rendered 1`] = `
<tbody>
  <Component
    key="1"
    member={
      Object {
        "avatarUrl": "http://testavatar.png",
        "id": 1,
        "name": "test field login",
      }
    }
  />
  <Component
    key="2"
    member={
      Object {
        "avatarUrl": "http://testavatar(1).png",
        "id": 2,
        "name": "test field login (1)",
      }
    }
  />
  <Component
    key="3"
    member={
      Object {
        "avatarUrl": "http://testavatar(2).png",
        "id": 3,
        "name": "test field login (2)",
      }
    }
  />
</tbody>
`;
